mdtraj/core/topology.py

```diff
@@ -641,13 +645,20 @@ def from_dataframe(
                     atom["resSeq"],
                     atom["segmentID"],
                 )
-
+            fc = atom["formal_charge"]
+            if fc is not None:
+                if pd.notna(fc):
+                    fc = float(fc)
```

Comment:
Formal charges are integers by the PDB definition. I understand not wanting to cast as `int` in case someone has input a floating point number, but if something like `1` is cast as a float here it will cause problems down the line if you try to write a PDB (ie it will fail because the line is too long).

I think there should be a check here to make sure the charges can be cast as `int`, and then they should be cast as `int` rather than float.

Reply:
Yes, I agree casting to int makes more sense. I've fixed the incorrect type annotation for the formal charge in atom and also added a check to make sure the charge can be  cast as `int`